native/core/src/execution/jni_api.rs

```diff
@@ -407,6 +560,20 @@ pub extern "system" fn Java_org_apache_comet_Native_releasePlan(
 ) {
     try_unwrap_or_throw(&e, |_| unsafe {
         let execution_context = get_execution_context(exec_context);
+        if execution_context.memory_pool_config.pool_type == MemoryPoolType::FairSpillTaskShared {
```

Comment:
Why we don't need to remove the pool for `MemoryPoolType::GreedyTaskShared` from `TASK_SHARED_MEMORY_POOLS`?

Reply:
It should be removed. I'll fix it in a follow-up commit.